src/load/azext_load/data_plane/utils/utils.py

```diff
@@ -0,0 +1,531 @@
+
+
+def download_file(url, file_path):
+    logger.debug("Downloading file started")
+    response = None
+    retries = 3
+    ex = None
+    while retries > 0:
+        try:
+            response = requests.get(url, stream=True, allow_redirects=True, timeout=60)
+            break
+        except Exception as e:  # pylint: disable=broad-except
+            ex = e
+            retries -= 1
+            logger.debug(
+                "Exception occurred while downloading file: %s. Retrying the request. Retries remaining: %d",
+                str(ex),
+                retries,
+            )
+    if retries == 0:
+        msg = f"Request for {url} failed after all retries: {str(ex)}"
+        logger.debug(msg)
+        raise FileOperationError(msg)
+
+    if response:
+        with open(file_path, "wb") as f:
+            for chunk in response.iter_content(chunk_size=1024):
+                if chunk:  # ignore keep-alive new chunks
+                    f.write(chunk)
+    logger.debug("Downloading file completed")
+
+
+def upload_file_to_test(client, test_id, file_path, file_type=None, wait=False):
+    logger.debug(
+        "Uploading file %s for the test %s with 'wait' %s",
+        file_path,
+        test_id,
+        "enabled" if wait else "disabled",
+    )
+    # pylint: disable-next=protected-access
+    file_path = validators._validate_path(file_path, is_dir=False)
+    with open(file_path, "rb") as file:
+        upload_poller = client.begin_upload_test_file(
+            test_id,
+            file_name=os.path.basename(file.name),
+            file_type=file_type,
+            body=file,
+        )
+        response = (
+            upload_poller.result()
+            if wait
+            else upload_poller.polling_method().resource()
+        )
+        logger.debug(
+            "Upload result for file with --wait%s passed: %s",
+            "" if wait else " not",
+            response,
+        )
+        return response
+
+
+def parse_cert(certificate):
+    logger.debug("Parsing certificate")
+    if len(certificate) != 1:
+        raise ValueError("Only one certificate is supported")
+    certificate = certificate[0]
+    name, value = certificate.get("name"), certificate.get("value")
+    # pylint: disable-next=protected-access
+    if not validators._validate_akv_url(value, "certificates"):
+        raise ValueError(f"Invalid AKV Certificate URL: {value}")
+    certificate = {
+        "name": name,
+        "type": "AKV_CERT_URI",
+        "value": value,
+    }
+    logger.debug("Parsed certificate: %s", certificate)
+    logger.debug("Certificate parsed successfully")
+    return certificate
+
+
+def parse_secrets(secrets):
+    logger.debug("Parsing secrets")
+    secrets_dict = {}
+    for secret in secrets:
+        name, value = secret.get("name"), secret.get("value")
+        if name is None or value is None:
+            raise ValueError("Both name and value are required for secret")
```

Comment:
Please use error types in https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md#error-type

Reply:
updated all the generic Error to specific error as mentioned in the guidelines